src/app/src/pages/Login.js

```diff
@@ -0,0 +1,91 @@
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { Box, Button, Input, Text, VStack } from '@chakra-ui/react';
+
+import { URLS } from '../constants';
+import { login } from '../store/authSlice';
+
+// https://stackoverflow.com/questions/39254562/csrf-with-django-reactredux-using-axios
+axios.defaults.xsrfHeaderName = 'X-CSRFTOKEN';
+axios.defaults.xsrfCookieName = 'csrftoken';
+axios.defaults.withCredentials = true;
+
+export default function Login() {
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const [errorDetail, setErrorDetail] = useState('');
+    const dispatch = useDispatch();
+    const navigate = useNavigate();
+
+    const loginRequest = () => {
+        axios
+            .post(URLS.LOGIN, {
+                email,
+                password,
+            })
+            .then(() => {
+                dispatch(login());
+            })
+            .catch(axiosError => {
+                setErrorDetail(axiosError.response.data?.detail);
+            });
+    };
+
+    const signedIn = useSelector(state => state.auth.signedIn);
+    const locationBeforeAuth = useSelector(
+        state => state.auth.locationBeforeAuth
+    );
+
+    useEffect(() => {
```

Comment:
Since `useEffect`s are so open ended and malleable, consider adding a comment block above this to describe what it does.

Reply:
df9283b0dab78e9a10a342aa6eb3bd6ca07a97a4